test_runner/fixtures/metrics.py

```diff
@@ -13,7 +13,8 @@ def __init__(self, name: str = ""):
         self.metrics = defaultdict(list)
         self.name = name
 
-    def query_all(self, name: str, filter: Dict[str, str]) -> List[Sample]:
+    def query_all(self, name: str, filter: Optional[Dict[str, str]] = None) -> List[Sample]:
+        filter = filter or {}
```

Comment:
Could the default value be '{}' directly? I.e.
```
 def query_all(self, name: str, filter: Dict[str, str] = {}) -> List[Sample]:
```

Reply:
Technically it can, but using mutable data structure as a default value could shoot us in the foot

Reply:
Yep, I learned this from @vadim2404 a few months back https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/